src/error.rs

```diff
@@ -132,26 +133,122 @@ pub trait ErrorDomain: Copy {
 }
 
 /// Generic error used for functions that fail without any further information
+#[macro_export]
+macro_rules! glib_bool_error(
+// Plain strings
+    ($msg:expr) =>  {
+        $crate::BoolError::new($msg, file!(), module_path!(), line!())
+    };
+
+// Format strings
+    ($($msg:tt)*) =>  { {
+        $crate::BoolError::new(format!($($msg)*), file!(), module_path!(), line!())
+    }};
+);
+
+#[macro_export]
+macro_rules! glib_result_from_glib(
+// Plain strings
+    ($ffi_bool:expr, $msg:expr) =>  {
+        $crate::BoolError::from_glib($ffi_bool, $msg, file!(), module_path!(), line!())
+    };
+
+// Format strings
+    ($ffi_bool:expr, $($msg:tt)*) =>  { {
+        $crate::BoolError::from_glib(
+            $ffi_bool,
+            format!($($msg)*),
+            file!(),
+            module_path!(),
+            line!(),
+        )
+    }};
+);
+
 #[derive(Debug)]
-pub struct BoolError(pub &'static str);
+pub struct BoolError {
+    pub message: Cow<'static, str>,
+    pub(crate) filename: &'static str,
+    pub(crate) function: &'static str,
+    pub(crate) line: u32,
```

Comment:
You need to make these public or otherwise the macro will not work when called from other crates.

You can make them ignored for the docs though :)